kola/harness.go

```diff
@@ -95,16 +95,14 @@ func (res resultSlice) ToTAP(w io.Writer) error {
 
 	for i, r := range res {
 		t := r.test
-		err := r.result
-		if err != nil {
-			switch err.(type) {
-			case skip.Skip:
-				_, werr = fmt.Fprintf(w, "ok %d - %s # SKIP: %s\n", i+1, t.Name, err)
-			default:
-				_, werr = fmt.Fprintf(w, "not ok %d - %s # %s\n", i+1, t.Name, err)
-			}
-		} else {
+
+		switch r.result.(type) {
+		case nil:
 			_, werr = fmt.Fprintf(w, "ok %d - %s\n", i+1, t.Name)
+		case skip.Skip:
+			_, werr = fmt.Fprintf(w, "ok %d - %s # SKIP: %s\n", i+1, t.Name, err)
+		case error:
```

Comment:
the type should be `error` to begin with so keeping this case as `default:` seems better, less likely for refactoring in the future to accidentally leave us with no log message at all.